src/main/java/org/elasticsearch/index/merge/MergeStats.java

```diff
@@ -176,5 +235,10 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeVLong(current);
         out.writeVLong(currentNumDocs);
         out.writeVLong(currentSizeInBytes);
+        if (out.getVersion().onOrAfter(Version.V_2_0_0)) {
```

Comment:
Upgrading from 1.x to 2.x will require a full cluster restart, so a 2.x node will never chat with a < 2.x node. Therefor the version checks are redundant and can be removed. (assuming that this change only goes into master)